gtsfm/utils/plane.py

```diff
@@ -0,0 +1,41 @@
+"""Utilities for generating data on planar surfaces.
+
+Authors: Ayush Baid
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def sample_points_on_plane(
+    plane_coefficients: Tuple[float, float, float, float],
+    range_x_coordinate: Tuple[float, float],
+    range_y_coordinate: Tuple[float, float],
+    num_points: int,
+) -> np.ndarray:
+    """Sample random points on a 3D plane ax + by + cz + d = 0.
+
+    Args:
+        plane_coefficients: coefficients (a,b,c,d) of the plane equation.
+        range_x_coordinate: desired range of the x coordinates of samples.
+        range_y_coordinate: desired range of the y coordinates of samples.
```

Comment:
This is kind of weird, how is this used? I'd imagine a use for limiting the range within the plane's frame of reference, but in the global frame, not so much. The points could be very far apart, depending on the slope of the plane.

Reply:
so the main way we're using it is by assuming that z is the only truly free variable, and x and y should be sampled within some range.